File: SozlukSwift/SozlukSwift/viewmodel/KayitViewModel.swift
//
//  KayitViewModel.swift
//  SozlukSwift
//
//  Created by Alperen Saraç on 9.08.2025.
//

import Foundation
final class KayitViewModel {
    func register(username: String, password: String, email: String, completion: @escaping (SimpleResponse) -> Void) {
        SozlukDao.shared.register(username: username, password: password, email: email) { result in
            DispatchQueue.main.async {
                switch result {
                case .success(let response):
                    completion(response)
                case .failure:
                    completion(SimpleResponse(success: false, message: "Bağlantı hatası"))
                }
            }
        }
    }
}

File: sozluk_flutter/lib/viewmodel/kayit_viewmodel.dart
import 'package:flutter/foundation.dart';

import '../dao/sozluk_dao.dart';
import '../model/simple_response.dart';


class KayitViewModel extends ChangeNotifier {
  final SozlukDao _dao;

  KayitViewModel({SozlukDao? dao}) : _dao = dao ?? SozlukDao();

  SimpleResponse? registerResult;
  bool loading = false;

  Future<void> register({
    required String username,
    required String password,
    required String email,
  }) async {
    loading = true;
    registerResult = null;
    notifyListeners();

    try {
      registerResult = await _dao.register(
        username: username,
        password: password,
        email: email,
      );
    } catch (_) {
      registerResult = SimpleResponse(success: false, message: 'Bağlantı hatası');
    }

    loading = false;
    notifyListeners();
  }
}
